packages/@aws-cdk/aws-events-targets/lib/ssm-automation.ts

```diff
@@ -0,0 +1,97 @@
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import * as ssm from '@aws-cdk/aws-ssm';
+import * as cdk from '@aws-cdk/core';
+import { addToDeadLetterQueueResourcePolicy, bindBaseTargetConfig, singletonEventRole, TargetBaseProps } from './util';
+
+/**
+ * Create an SSM Automation Event Target
+ */
+export interface SsmAutomationProps extends TargetBaseProps {
+  /**
+   * Role to be used to run the Document
+   *
+   * @default - a new role is created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The input parameters for the Automation document.
+   *
+   * @default - no input parameters passed to the document
+   */
+  readonly input?: { [key: string]: string[] };
+
+  /**
+   * Role to be used to run the Automation on your behalf. This should be a role
+   * that allows the Automation service principal (ssm.amazonaws.com) to assume
+   * and run the actions in your Automation document. Only required if the
+   * document type is `Automation`.
+   *
+   * @default - no role assumed
+   */
+  readonly ssmAssumeRole?: iam.IRole;
+}
+
+/**
+ * Create an SSM Automation Event Target
+ */
+export class SsmAutomation implements events.IRuleTarget {
+  private documentArn: string;
+
+  constructor(
+    /**
+     * Can be an instance of `ssm.CfnDocument` or a share/managed document ARN.
+     */
+    public readonly document: ssm.CfnDocument | string,
```

Comment:
In fact, I don't think we want to use the Cfn type of anything here. When Document gets implemented, this will then create a breaking change. This should use an `IDocument` as the type. I realize that this isn't implemented yet but I think that it needs to be before we can accept this change.

Reply:
I had the exact same thought and started down that road.. but figured I would get your feedback first. Is there any existing or in-progress work already on implementing L2 constructs for `Document`? Is it possible to just wrap `CfnDocument` to avoid the breaking change down the road?

Reply:
I'd be willing to start the implementation of Document. Is there any documentation, standards, or requirements I can follow or reference for that implementation? Other than looking at other similar L2 implementations.

Reply:
One more thought.. once Document is implemented, there would be no need for the union type as theoretically (and hopefully) there would be some sort of `.fromLookup()` method on `Document` for shared documents.

Reply:
See this package: https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/example-construct-library

Specifically, the example resource, as that would be what the Document would be. Please also open that in a separate PR.